jupyter_forward/core.py

```diff
@@ -136,11 +142,13 @@ def start(self):
         # Logfile will be in $TMPDIR if defined on the remote machine, otherwise in $HOME
 
         try:
-            check_jupyter_status = 'command -v jupyter'
+            check_jupyter_status = 'sh -c "command -v jupyter"'
             if self.conda_env:
-                check_jupyter_status = f'conda activate {self.conda_env} && command -v jupyter'
+                check_jupyter_status = (
+                    f'conda activate {self.conda_env} && sh -c "command -v jupyter"'
+                )
             self._jupyter_info(check_jupyter_status)
-            if self.dir_exists('$TMPDIR'):
+            if self.envvar_exists('TMPDIR') and self.dir_exists('$TMPDIR'):
                 self.log_dir = '$TMPDIR'
             else:
                 self.log_dir = '$HOME'
```

Comment:
We should double check to make sure `$HOME` exists and is pointing to a directory too

Reply:
If `$HOME` does not exist, should we abort? I could do something like

```py3
            if self.envvar_exists('TMPDIR') and self.dir_exists('$TMPDIR'):
                self.log_dir = '$TMPDIR'
            elif self.envvar_exists('HOME') and self.dir_exists('$HOME'):
                self.log_dir = '$HOME'
            else:
                console.log(
                    '[bold red]Can not determine directory for log file'
                )
                sys.exit(1)

```

Reply:
And if both `$TMPDIR` and `$HOME` are invalid, we should probably raise an error

Reply:
For a more informative error, we can be more verbose in `Can not determine directory for log file'` by specifying that both `$TMPDIR` and `$HOME` were found to be undefined or pointing to non-existing directories....